AppLovin MAX Demo App - Java/app/src/main/java/com/applovin/enterprise/apps/demoapp/data/main/AdType.java

```diff
@@ -10,13 +10,27 @@
 public class AdType
         implements ListItem
 {
-    private final String                    adType;
+    public interface OnTap
+    {
+        public void onTap();
+    }
+
+    private final String adType;
     private final Intent intent;
+    private final OnTap  onTap;
 
     public AdType(final String adType, final Intent intent)
     {
         this.adType = adType;
         this.intent = intent;
+        this.onTap = null;
+    }
+
+    public AdType(final String adType, final OnTap onTap)
```

Comment:
This is what the `Runnable` interface in Java is for:

```suggestion
    public AdType(final String adType, final Runnable runnable)
```